obspy/imaging/tests/test_scan.py

```diff
@@ -172,11 +172,35 @@ def test_multiple_sampling_rates(self):
                 fp.flush()
                 fp.seek(0)
                 files.append(fp.name)
-            with ImageComparison(self.path, 'scan_mult_sampl.png')\
-                    as ic:
+
+            # make image comparison instance and set manual rms (see #2089)
+            image_comp = ImageComparison(self.path, 'scan_mult_sampl.png')
+            image_comp.tol = 40
+
+            with image_comp as ic:
+
+                obspy_scan(files + ['--output', ic.name, '--print-gaps'])
+
                 with CatchOutput() as out:
                     obspy_scan(files + ['--output', ic.name, '--print-gaps'])
-                self.assertEqual(expected, out.stdout.splitlines())
+
+                # read output and compare with expected
+                # only check if datetime objects are close, not exact
+                output = out.stdout.splitlines()
+                for ex_line, out_line in zip(expected, output):
+                    ex_split = ex_line.split(' ')
+                    out_split = out_line.split(' ')
+                    for ex_str, out_str in zip(ex_split, out_split):
+                        try:
+                            utc1 = UTCDateTime(ex_str)
+                            utc2 = UTCDateTime(out_str)
+                        except (ValueError, TypeError):
+                            # if str is not a datetime it should be equal
+                            self.assertEqual(ex_str, out_str)
+                        else:
+                            # datetimes just need to be close
+                            t1, t2 = utc1.timestamp, utc2.timestamp
+                            self.assertTrue(abs(t1 - t2) < .001)
```

Comment:
I'm a bit puzzled why we have to change this unit test.. there were some PRs with changes to UTCDateTime, are those responsible for needing this change?

Reply:
@megies, no I thought that at first too but it is actually due to a change in how matplotlib handles rounding (see [this comment](https://github.com/obspy/obspy/pull/2089#issuecomment-375124108)). This test needed to change in order to not simply compare the string representation of the  matplotlib output (which for some reason is version dependent) but rather to test that the values returned are very close to the expected result.

Reply:
Oh.. I got it now. Thanks for clarification. So the reason is that the data during `obspy-scan` goes through a `date2num`-`num2date` cycle and our "expected" data is sitting in the middle, created with an older matplotlib than is used in CI.

This also implies that data saved by obspy-scan on older mpl versions gets deserialized with slightly different results when using a newer matplotlib.. :-/